docs/build-protocol-info.md

```diff
@@ -0,0 +1,168 @@
+---
+id: build-protocol-info
+title: Polkadot Protocol Information
+sidebar_label: Polkadot Protocol
+---
+
+This page serves as a high-level introduction to the Polkadot protocol with terminology that may be
+specific to Polkadot, notable differences to other chains that you may have worked with, and
+practical information for dealing with the chain.
+
+## DOT Tokens
+
+- **Token decimals:** 12
+- **Balance type:** `u128`
+
+## Addresses
+
+In Polkadot (and most Substrate chains), user accounts are identified by a 32-byte (256-bit)
+`AccountId`. This is simply the public key for the x25519 cryptography used by Substrate.
+
+Polkadot (and Substrate) use the SS58 address format. This is a broad "meta-format" designed to
+handle many different cryptographies and chains. It has much in common with Bitcoin's Check58
+format such as a version prefix, a hash-based checksum suffix, and base-58 encoding.
+
+See the [SS58 page](https://github.com/paritytech/substrate/wiki/External-Address-Format-(SS58)) in
+the Substrate wiki for encoding information.
+
+### Cryptography
+
+Polkadot supports the following cryptographic key pairs and signing algorithms:
+
+- Ed25519
+- Sr25519 - Schorr signatures on the Ristretto group
+- ECDSA signatures on secp256k1
+
+Note that the address for a secp256k1 key is the SS58 encoding of the _hash of the public key_ in
+order to reduce the public key from 33 bytes to 32 bytes.
+
+## Existential Deposit
+
+Polkadot uses an _existential deposit_ (ED) to prevent dust accounts from bloating state. If an
+account drops below the ED, it will be _reaped,_ i.e. completely removed from storage and the nonce
+reset.
+
+Wallets and custodians who track account nonces for auditing purposes should take care not to have
+accounts reaped, as users could refund the address and try making transactions from it. The
+Balances pallet provides a `transfer_keep_alive` function that will return an error and abort
+rather than make the transfer if it would result in account reaping.
+
+## Free vs. Reserved vs. Locked vs. Vesting Balance
+
+Account balance information is stored in
+[`AccountData`](https://substrate.dev/rustdocs/master/pallet_balances/struct.AccountData.html).
+Polkadot primarily deals with two types of balances: free and reserved.
+
+For most operations, free balance is what you are interested in. It is the "power" of an account in
+staking and governance, for example. Reserved balance represents funds that have been set aside by
+some operation and still belong to the account holder, but cannot be used.
+
+Locks are an abstraction over free balance that prevent spending for certain purposes. Several
```

Comment:
Would describe the different kinds of actions that result in locks: staking, referenda, council, etc